src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs

```diff
@@ -37,13 +37,15 @@ public partial class HostBuilder : IHostBuilder
         private HostingEnvironment? _hostingEnvironment;
         private IServiceProvider? _appServices;
         private PhysicalFileProvider? _defaultProvider;
+        private readonly bool _defaultProviderFactoryUsed;
 
         /// <summary>
         /// Initializes a new instance of <see cref="HostBuilder"/>.
         /// </summary>
         public HostBuilder()
         {
             _serviceProviderFactory = new ServiceFactoryAdapter<IServiceCollection>(new DefaultServiceProviderFactory());
+            _defaultProviderFactoryUsed = true;
```

Comment:
Where is this ever set to false?  Should that be done in `UseServiceProviderFactory`?

Reply:
Yes, it seems to me that this field should be set to false when `UseServiceProviderFactory` is called. This was an oversight.